env.py

```diff
@@ -0,0 +1,111 @@
+from __future__ import print_function
+
+import argparse
+import glob
+import os
+import sys
+import textwrap
+import subprocess
+
+
+def ensure_dir(path):
+    if os.path.exists(path):
+        return
+    
+    os.mkdir(path)
+    return path
+
+
+def traverse_toolsets(f):
+    msbuild_dir = r'C:\Program Files (x86)\MSBuild'
+    toolsets = glob.glob(r'{}\Microsoft.Cpp\v4.0\*\Platforms\*\PlatformToolsets\*'.format(msbuild_dir))
+    for toolset in toolsets:
+        f(toolset)
+
+
+def clcache_props_path(toolset):
+    return os.path.join(toolset, 'ImportAfter', 'Clcache.props')
+
+
+def generate_props_content(clcache_dir):
+    return textwrap.dedent('''
+        <Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+          <PropertyGroup>
+            <ExecutablePath>{clcache_dir};$(ExecutablePath)</ExecutablePath>
+          </PropertyGroup>
+        </Project>''').format(clcache_dir=clcache_dir)[1:]
+
+
+def check_call_quiet(*args, **kwargs):
+    with open(os.devnull, 'w') as devnull:
+        kwargs['stderr'] = devnull
+        kwargs['stdout'] = devnull
+        subprocess.check_call(*args, **kwargs)
+
+
+def set_system_variable(var, value):
+    if value:
+        check_call_quiet(['setx', '-m', var, value])
+    else:
+        reg_path = r'HKLM\SYSTEM\CurrentControlSet\Control\Session Manager\Environment'
```

Comment:
Would it also be sufficient to affect just the current user, i.e. use `HKCU` instead of `HKLM`?